apps/web/src/components/templates/TemplateSettings.tsx

```diff
@@ -1,16 +1,37 @@
 import { Grid, useMantineColorScheme } from '@mantine/core';
-import { colors } from '../../design-system';
+import { Button, colors } from '../../design-system';
 import { NotificationSettingsForm } from './NotificationSettingsForm';
 import { TemplatesSideBar } from './TemplatesSideBar';
 import { TriggerSnippetTabs } from './TriggerSnippetTabs';
 import styled from '@emotion/styled';
 import { useTemplateController } from './use-template-controller.hook';
 import { ActivePageEnum } from '../../pages/templates/editor/TemplateEditorPage';
+import { Trash } from '../../design-system/icons';
+import { useState } from 'react';
+import { DeleteConfirmModal } from './DeleteConfirmModal';
+import { useNavigate } from 'react-router-dom';
 
 export const TemplateSettings = ({ activePage, setActivePage, showErrors, templateId }) => {
   const { colorScheme } = useMantineColorScheme();
 
-  const { editMode, template, trigger } = useTemplateController(templateId);
+  const { editMode, template, trigger, deleteTemplate } = useTemplateController(templateId);
+  const [toDelete, setToDelete] = useState(false);
+
+  const navigate = useNavigate();
+
+  const confirmDelete = () => {
+    navigate('/templates');
+    deleteTemplate();
```

Comment:
Maybe worth awaiting this here in case an exception will be thrown? We can also add a loading state for the button? We can use mutate from react query for it. And display an error using the message from mantine. Wdyt?